### PR TITLE
refactor(InputExtraActions): allow any ReactElement as child

### PR DESCRIPTION
**Pull request by ryancarville**
## Definition of Done
- [ ] User interface and experience have been verified by a designer
- [x] I have added thorough Cypress tests (All properties and interactions have been tested).
- [x] Cross-browser compatibility - The component look consistent in the latest version of Chrome, Safari, Firefox and Edge.
- [x] I've added documentation in Storybook. all properties are reflected in table and controls.
- [x] User interface is compliant with WCAG 2.1 AA. Ensure these items are fulfilled:
    - [x] Keyboard operability: All functionality can be accessed using only the keyboard
    - [x] Logical tab order: Interactive elements follow a consistent and logical tab order
    - [x] Visual indication of focus: Focused elements are visually highlighted for keyboard navigation
    - [x] Color contrast: Sufficient contrast is used between text and background, as well as form fields and background
    - [x] Alternative visual cues: Information is not solely reliant on color and includes alternative visual indicators
    - [x] Semantic markup: Headings, lists, and form elements are marked up with appropriate semantic tags

### CODE REVIEW

File: src/types/input.ts [hunk 1]
```diff
@@ -51,9 +52,7 @@ export type InputActionsProps = {
 };
 
 export type ExtraAction = {
-    icon: ReactElement;
-    title: string;
-    callback: () => void;
+    tooltip: TooltipProps;
```
Comment (imoutaharik) [inline src/types/input.ts:55]: Are we enforcing the use of Tooltip or just provide the option to add one if we want to?
Content on the Tooltip is required, I'm wondering what would be usage guidelines here => Actions buttons should always have a tooltip?

Comment (ryancarville) [reply src/types/input.ts:55]: Currently the structure is that any `extraAction` is a `Tooltip` with any `child` element they wish to use.  This is what is reflected in the Figma file.  But can also modify this to allow for more flexibility.

Comment (imoutaharik) [reply src/types/input.ts:55]: I see. Then that's fine but we could anyway double-check and get confirmation with the Designers.

Comment (ryancarville) [reply src/types/input.ts:55]: OK will raise it at today's sync then can proceed from there.

Comment (ryancarville) [reply src/types/input.ts]: @imoutaharik I have made the updates to allow a tooltip or element.  Interested in your thoughts on the implementation.

File: src/utilities/input.tsx [hunk 2]
```diff
@@ -137,24 +145,35 @@ export const InputActions = ({
...
-        );
+        if (tooltip) {
+            return (
+                <Tooltip
+                    content={tooltip.content}
+                    key={`extra-action-tooltip-${id}`}
+                    disabled={tooltip.disabled ?? disabled}
+                    data-test-id={`${dataTestId}-extra-action`}
+                >
+                    {cloneElement(tooltip.children, { disabled: tooltip.disabled ?? disabled })}
```
Comment (imoutaharik) [inline src/utilities/input.tsx:164]: Is it the reason why we choose between Tooltip and Element? Like the whole Tooltip doesn't work if passed as ReactElement?

Comment (ryancarville) [reply src/utilities/input.tsx]: @imoutaharik I am update this to allow any `ReactElement`.

**Pull request closed without merging**